p2p/src/peer_manager/peerdb/store.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::Duration;
+
+pub trait PeerDbStore: Send {
```

Comment:
This interface is overly simplistic. It doesn't support transactions too. I see that in the implementation you have transactions all over. There should be a way to start transaction through this interface, then use that to do these operations.

Reply:
Transactions added. I used chainstate storage as a base, but simplified it (to avoid using macros).
`PeerManager` will now stop working if a storage error is detected (for example if a commit fails).